pkg/bff/credentials.go

```diff
@@ -0,0 +1,150 @@
+package bff
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	jwt "github.com/golang-jwt/jwt/v4"
+	"github.com/trisacrypto/directory/pkg/bff/config"
+	"github.com/trisacrypto/directory/pkg/gds/admin/v2"
+	"github.com/trisacrypto/directory/pkg/gds/tokens"
+)
+
+// Credentials implements the admin.Credentials interface to provide access and refresh
+// tokens to the GDSClient.
+type Credentials struct {
```

Comment:
One thing that I didn't understand off the bat was that this implements the `admin.Credentials` interface and that's what ensures that the client can renew their credentials via the [`checkAuthentication`](https://github.com/trisacrypto/directory/blob/main/pkg/gds/admin/v2/client.go#L767) method. Could you please document this in the doc string and also add a:

```
// Ensure Credentials implements the admin.Credentials interface
var _ admin.Credentials = &Credentials{}
```